apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -985,6 +985,14 @@ func (m *ModelManagerAPI) getModelInfo(tag names.ModelTag, withSecrets bool, wit
 		modelAdmin = err == nil
 	}
 
+	// If the user is not model admin, they must at least have read access to
+	// the model.
+	if !modelAdmin {
+		if err = m.authorizer.HasPermission(permission.ReadAccess, model.ModelTag()); err != nil {
+			return params.ModelInfo{}, errors.Trace(apiservererrors.ErrPerm)
+		}
+	}
```

Comment:
This method is called during create model, but shouldn't activate this code.

The existing code isn't the cleanest to understand, esp. when the potential to get it wrong is leaking all the users and secrets. 

I'd rather the code got all the data everyone could see and then returned if it wasn't the correct level. Any model admin data could be added once that check is done. Could we potentially update the code to do that instead? That would give me more confidence that we weren't leaking anything by accident and make the method more robust.